packages/create-anywidget/create.js

```diff
@@ -1,70 +1,86 @@
-// @ts-check
 import * as fs from "node:fs/promises";
 import * as path from "node:path";
 import * as url from "node:url";
-
 import snakecase from "just-snake-case";
 
-let __dirname = path.dirname(url.fileURLToPath(import.meta.url));
+export async function create(target, options) {
+  const __dirname = path.dirname(url.fileURLToPath(import.meta.url));
+  const copyFrom = path.resolve(__dirname, options.template);
+  const copyTo = target;
+  const newName = snakecase(options.name); 
+
+  const allFiles = await gatherFiles(copyFrom);
+  const updatedFiles = renameGitignore(allFiles);
+  const updatedContentFiles = replaceWidgetName(updatedFiles, newName); // Pass newName as argument
+  const updatedPathFiles = updateFilePaths(updatedContentFiles, copyFrom, copyTo, newName); // Pass newName as argument
 
-/** @param {string} root */
-async function* walk(root) {
-	for (let entry of await fs.readdir(root, { withFileTypes: true })) {
-		if (entry.isFile()) {
-			yield {
-				type: "file",
-				path: path.resolve(root, entry.name),
-			};
-		} else if (entry.isDirectory()) {
-			yield {
-				type: "directory",
-				path: path.resolve(root, entry.name),
-			};
-			yield* walk(path.resolve(root, entry.name));
-		} else {
-			throw Error("unknown type");
-		}
-	}
+  console.log(copyTo);
+  try {
+    await createFiles(updatedPathFiles);
+    console.log("All files created successfully!");
+  } catch (err) {
+    console.error("Error writing files:", err);
```

Comment:
I'd like to avoid logging from these functions.

we should just let `createFiles` throw and error and then handle that error in `index.js` (e.g., `process.exit(1)`).

Reply:
```javascript
return createFiles(updatedPathFiles)
```

Reply:
or actually, why not write the files and then return an array of the written paths (if successful). Then we can (optionally) log those files in `index.js`.

```javascript
await writeFiles(updatedPathFiles);
return updatedPathFiles.map(file => file.path);
```